### ISSUE TITLE
plugin is making calls to external web sites without warning user

### ISSUE THREAD

**Issue by axb21** [mokeyish/obsidian-code-emitter#2]
I was poking through this plugin's code to see how it worked and came across this line in the backend for java:

https://github.com/mokeyish/obsidian-code-emitter/blob/7f2ad92c05d1ea1b4f19aadae5c35651f8df8cfa/src/backend/languages/java.ts#L3

It appears that this plugin makes a call to a third-party web site to compile code. It is unclear whether `sololearn.com` is OK with this use of their api, but in any case, the user should certainly be warned about sending potentially-sensitive source code for compilation to a third party they may not know or trust. Also, the plugin's README should be very clear about what's happening on the backend so that users can make an informed choice about whether to trust it.

Similar issues exist for the other compiled languages, it seems:

For `kotlin`:
https://github.com/mokeyish/obsidian-code-emitter/blob/7f2ad92c05d1ea1b4f19aadae5c35651f8df8cfa/src/backend/languages/kotlin.ts#L3

For `rust`:
https://github.com/mokeyish/obsidian-code-emitter/blob/7f2ad92c05d1ea1b4f19aadae5c35651f8df8cfa/src/backend/languages/rust.ts#L3

For `go`:
https://github.com/mokeyish/obsidian-code-emitter/blob/7f2ad92c05d1ea1b4f19aadae5c35651f8df8cfa/src/backend/languages/go.ts#L3

**Comment by axb21** [comment]
Hi @eleanorkonik : I saw this plugin listed in your Obsidian Roundup newsletters (which I love by the way!) so I thought you might be interested in this issue I just posted about it.

Personally, I am not OK with plugins making calls to third-party web sites without alerting me that they are going to do this or at least describing in the README that this is how they function. The jupyter plugin runs a local jupyter notebook and makes it calls to that, for instance, which they describe in their README. I was a bit disappointed that this particular plugin does not do that, and that it was listed in the roundup without some kind of warning.
